### PR TITLE
[Bug/Doc] Add documentation requiremets and loss method documentation

### PR DESCRIPTION
**Pull request by vertefra**
- Update documentation adding description for `loss` attribute
- Update default `QCOG` version
- Add code generator dependency in `docs` requirements

### CODE REVIEW

File: qcog_python_client/qcog.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def get(self, *args: Any, **kwargs: Any) -> Coroutine | dict:
 SUCCESS_STATUS = (TrainingStatus.completed,)
 
 
-DEFAULT_QCOG_VERSION = "0.0.74"
+DEFAULT_QCOG_VERSION = "0.0.89"
```
Comment (jamesETsmith) [inline qcog_python_client/qcog.py:70]: Are you bumping this manually or do we have a way to bump it automatically?

File: qcog_python_client/qcog.py [hunk 2]
```diff
@@ -591,7 +591,7 @@ def status(self) -> TrainingStatus:
         )
 
         self.last_status = TrainingStatus(self.status_resp["status"])
-
+        print("Status response: ", self.status_resp)
```
Comment (jamesETsmith) [inline qcog_python_client/qcog.py:594]: Do we have the logger in here yet?

File: tests/test_example_2.py [hunk 3]
```diff
@@ -136,10 +136,10 @@ def test_train_two():
         batch_size=len(df_train),
         num_passes=10,
         weight_optimization=AnalyticOptimizationParameters(),
-        get_states_extra=LOBPCGFastStateParameters(iterations=10),
+        get_states_extra=LOBPCGFastStateParameters(iterations=20),
     )
 
-    qcml.wait_for_training()
+    qcml.wait_for_training(poll_time=10)
```
Comment (jamesETsmith) [inline tests/test_example_2.py:142]: Why the `poll_time`?

**Review by jamesETsmith — CHANGES_REQUESTED** [review]

File: qcog_python_client/qcog.py (hunk 1 above)
Comment (vertefra) [reply qcog_python_client/qcog.py:70]: The bump is manual. There is an ongoing discussion about how to change the way that specify the in-use qcog package from the client

File: tests/test_example_2.py (hunk 3 above)
Comment (vertefra) [reply tests/test_example_2.py:142]: It's a name parameter that has a default value of 60 seconds. During the examples, reducing the poll time to 10 seconds on short trainings, is more effective in retrieving metrics about the training progress

**Review by jamesETsmith — APPROVED** [review]

**Pull request merged**